### PR TITLE
Chart

### PR DESCRIPTION
**Pull request by smilenow24**
Issue: Analytics with calculations and charts

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,32 +23,66 @@ export default function App() {
     month: '2-digit',
     day: '2-digit',
   });
+  const differenceDate = inputs[0].date.getTime() - inputs[inputs.length - 1].date.getTime();
+  const differenceDays = Math.round(differenceDate / (24 * 3600 * 1000)) + 1;
+  const totalBudget = 30000;
+  const dailyTotalBudget = 1000;
+  const saldoIncrease = inputs[0].increase - dailyTotalBudget;
```
Comment (macfrei) [inline src/App.js:31]: I noticed that your `App.js` is getting bigger and bigger and is not well readable. I would highly recommend you to refactor your this file.

File: src/components/InputDataDialog.js [hunk 2]
```diff
@@ -1,54 +1,50 @@
...
+      <button>Submit</button>
+    </FormInputData>
+  );
+
+  function handleSubmit(inputEvent) {
+    inputEvent.preventDefault();
+    const inputData = inputEvent.target;
+    const inputDataValue = inputData.elements.inputfield.value.trim();
+
+    updateInput(inputDataValue);
```
Comment (macfrei) [inline src/components/InputDataDialog.js]: I would recommend to use more specific names. _What_ are you updating exactly? A better name would probably be something like: `updateEnergyConsumption`.

File: src/components/ReturnButton.spec.js [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 describe('ReturnButton', () => {
-
-    it('is an accessible form', () => {
-        render(<ReturnButton />);
-        screen.getByRole('button', {name: 'return to main site button'});
-        screen.queryAllByText('img', {src: 'iconReturnButton'});
-      });
-});
\ No newline at end of file
+  it('is an accessible form', () => {
```
Comment (macfrei) [inline src/components/ReturnButton.spec.js]:
```suggestion
  it('renders a button', () => {
```

**Review by macfrei — COMMENTED** [review]
Looks good, I would really recommend a refactor soon :)

File: src/App.js (hunk 1 above)
Comment (smilenow24) [reply src/App.js:31]: Hey Lene, thanks a lot for your good review: The refactoring fits well with my next userstory. Hope you can give me a hint how I can best proceed the refactoring. Greetings

**Pull request merged**